APEXMALHAR-1985: Using startRow to fetch startRowToken

DT-Priyanka opened the pull request.

DT-Priyanka committed:
APEXMALHAR-1985: Using startRow to fetch startRowToken

DT-Priyanka committed:
APEXMALHAR-1948: CassandraStore Should Allow You To Specify Protocol Version.

bhupeshchawda commented on contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraPOJOInputOperator.java:
```diff
@@ -359,24 +373,43 @@ public void emitTuples()
+
+  private Long fetchKeyTokenFromDB(Object keyValue)
+  {
+    PreparedStatement statement = store.getSession().prepare(TOKEN_QUERY);
```
Can we move this to setup()? This will get called for each tuple, so better to have a prepared statement created just once.

DT-Priyanka replied:
This logic is going to be removed by next pull request for cassandra scan, so not changing it.

bhupeshchawda replied:
Can you move the relevant changes from #186 to this one? In case this is ever cherry-picked, it may leave these changes as is, which might not be very efficient.

DT-Priyanka replied:
Once you merge #186, I will rebase this pull request against those changes. Is that okay?

bhupeshchawda commented on contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraPOJOInputOperator.java:
```diff
@@ -359,24 +373,43 @@ public void emitTuples()
+  private Long fetchKeyTokenFromDB(Object keyValue)
+  {
+    PreparedStatement statement = store.getSession().prepare(TOKEN_QUERY);
+    BoundStatement boundStatement = new BoundStatement(statement);
```
May be even this?

bhupeshchawda commented on contrib/src/test/java/com/datatorrent/contrib/cassandra/CassandraOperatorTest.java:
```diff
@@ -333,26 +333,26 @@ public void testCassandraInputOperator()
     inputOperator.emitTuples();
     inputOperator.endWindow();
-    Assert.assertEquals("rows from db", 14, sink.collectedTuples.size());
+    Assert.assertEquals("rows from db", 26, sink.collectedTuples.size());
```
Why is this 26?

DT-Priyanka replied:
This depends on hash value of primary key column value, that's why.

bhupeshchawda commented on contrib/src/test/java/com/datatorrent/contrib/cassandra/CassandraOperatorTest.java:
```diff
@@ -333,26 +333,26 @@ public void testCassandraInputOperator()
     inputOperator.activate(context);
 
-    inputOperator.setStartRow(1);
     inputOperator.setLimit(10);
```
Shouldn't setLimit() also be called before activate() ?

DT-Priyanka replied:
will change this.

Commit:
APEXMALHAR-2015: Projection Operator & its unit tests
 - Projection Operator
 - Unit tests for select/drop fields & projection operator

bhupeshchawda committed:
Merge branch 'APEXMALHAR-1948-cassandra-protocolVersion' of https://github.com/DT-Priyanka/incubator-apex-malhar

Commit:
Merge branch 'master' of https://github.com/pradeepdalvi/incubator-apex-malhar

ilooner committed:
APEXMALHAR-1991 #resolve #comment Move Dimensions Computation Classes to org.apache.apex.malhar package

otterc committed:
APEXMALHAR-2031 enable WindowDataManager to save window files under user configured path

ilooner committed:
Merge branch 'APEXMALHAR-2031'

otterc committed:
APEXMALHAR-2036 cleanup of intermediate test files

ilooner committed:
Merge branch 'APEXMALHAR-2036'
